logstash-core/lib/logstash/inputs/metrics.rb

```diff
@@ -44,9 +44,9 @@ def update(snapshot)
       #   - We can use a synchronization mechanism between the called thread (update method)
       #   and the plugin thread (run method)
       #   - How we handle back pressure here?
-      #   - one snashot should be only one event
-      snapshot.metric_store.to_events.each do |event|
-        @queue << event
+      #   - one snashot should be only one event ?
+      snapshot.metric_store.all_metrics.each do |metric|
```

Comment:
As I shared in the metrics store, I'm somehow in the fence for methods like this one. What do you think about a notation like

```
snapshot.metric_store.each do |metric|
   ...
end
```